internal/k8s/kind.go

```diff
@@ -26,22 +25,30 @@ import (
 // objects, so we have to use a type assertion to detect kinds that
 // we care about.
 func KindOf(obj interface{}) string {
-	switch obj := obj.(type) {
-	case *v1.Secret:
-		return "Secret"
-	case *v1.Service:
-		return "Service"
-	case *v1.Endpoints:
-		return "Endpoints"
-	case *v1beta1.Ingress:
-		return "Ingress"
-	case *projectcontour.HTTPProxy:
-		return "HTTPProxy"
-	case *projectcontour.TLSCertificateDelegation:
-		return "TLSCertificateDelegation"
-	case *unstructured.Unstructured:
-		return obj.GetKind()
-	default:
+	sch := scheme.Scheme
+	_ = projectcontour.AddToScheme(sch)
```

Comment:
this will help us add current contour types however I'm not sure how performant this approach is ? We may pass the schema from upper but does it create a lot of issue?

Reply:
This help us to not include each object type for contour (especially v1- v1beta1 etc in future maybe)

Reply:
I changed to addToSchema since it messed up concurrency in tests. However im still asking this question :smile: 

with this approach we dont need to add each type to our switch case (especially everytime Contour supports another type)

Reply:
Perhaps we could provide an 'all types registered' schema as an export of the `k8s` package - this would be useful in a number of places, and would be a good place to add more types as we go.

Reply:
Yep, +1 to exposing a universal Contour scheme from the `k8s` package.

Reply:
So which one should i go with it , exposing universal schema or create an issue for it :question:

Reply:
Let's get your change in, if you could log an issue for adding the schema, that would be great!

Reply:
https://github.com/projectcontour/contour/issues/2818
created a issue for universal Contour types